### PR TITLE
os/Transaction: dump alloc hint flags in op

### PR DESCRIPTION
**Pull request by Aran85**
Signed-off-by: Zengran Zhang <zhangzengran@sangfor.com.cn>

- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

**Comment by Aran85** [comment]
@tchaikov easy change, will you mind to review? Thanks!

File: src/os/Transaction.cc [hunk 1]
```diff
@@ -493,11 +493,13 @@ void Transaction::dump(ceph::Formatter *f)
         ghobject_t oid = i.get_oid(op->oid);
         uint64_t expected_object_size = op->expected_object_size;
         uint64_t expected_write_size = op->expected_write_size;
+        uint32_t alloc_hint_flags = op->alloc_hint_flags;
         f->dump_string("op_name", "op_setallochint");
         f->dump_stream("collection") << cid;
         f->dump_stream("oid") << oid;
         f->dump_stream("expected_object_size") << expected_object_size;
         f->dump_stream("expected_write_size") << expected_write_size;
+        f->dump_stream("alloc_hint_flags") << alloc_hint_flags;
```
Comment (tchaikov) [inline src/os/Transaction.cc]: why do you prefer using `dump_stream()` over `dump_unsigned()`? how about dump the flags using `dump_format()` in hex? as this field is or'ed flags of `CEPH_OSD_ALLOC_HINT_FLAG_*`

Comment (Aran85) [reply src/os/Transaction.cc]: Thanks, inspired by your comment I changed the dump format, acceptable?

File: src/os/Transaction.cc [hunk 2]
```diff
@@ -493,11 +493,13 @@ void Transaction::dump(ceph::Formatter *f)
         ghobject_t oid = i.get_oid(op->oid);
         uint64_t expected_object_size = op->expected_object_size;
         uint64_t expected_write_size = op->expected_write_size;
+        uint32_t alloc_hint_flags = op->alloc_hint_flags;
         f->dump_string("op_name", "op_setallochint");
         f->dump_stream("collection") << cid;
         f->dump_stream("oid") << oid;
         f->dump_stream("expected_object_size") << expected_object_size;
         f->dump_stream("expected_write_size") << expected_write_size;
+        f->dump_stream("alloc_hint_flags") << ceph_osd_alloc_hint_flag_string(alloc_hint_flags);
```
Comment (tchaikov) [inline src/os/Transaction.cc]: great! can we use `dump_string()`? it is more light-weighted than `dump_stream()`.

Comment (Aran85) [reply src/os/Transaction.cc]: Thanks for pointed it out, I did not realized that.

**Pull request merged**